core/domain/exp_services_test.py

```diff
@@ -4085,10 +4085,45 @@ def test_update_interaction_solutions(self):
             })], 'Changed interaction_solutions.')
 
         exploration = exp_fetchers.get_exploration_by_id(self.NEW_EXP_ID)
+
         self.assertEqual(
             exploration.init_state.interaction.solution.to_dict(),
             solution)
 
+    def test_update_solution_no_interaction_id(self):
+        exp_services.update_exploration(
+            self.albert_id, self.NEW_EXP_ID,
+            [exp_domain.ExplorationChange({
+                'cmd': exp_domain.CMD_ADD_STATE,
+                'state_name': 'New state'
+            })],
+            'Add new_state state'
+        )
+
+        solution = {
+            'answer_is_exclusive': False,
+            'correct_answer': 'helloworld!',
+            'explanation': {
+                'content_id': 'solution',
+                'html': '<p>hello_world is a string</p>'
+            },
+        }
+
+        invalid_solution_changes = [exp_domain.ExplorationChange({
+            'cmd': exp_domain.CMD_EDIT_STATE_PROPERTY,
+            'property_name': exp_domain.STATE_PROPERTY_INTERACTION_SOLUTION,
+            'state_name': 'New state',
+            'new_value': solution
+        })]
+
+        with self.assertRaises(KeyError):
```

Comment:
Typically, the error message is also asserted.. Is any error message raised along with the keyerror? If so assert that too...